sway/desktop/xdg_shell.c

```diff
@@ -248,7 +249,8 @@ void handle_xdg_shell_surface(struct wl_listener *listener, void *data) {
 	wlr_log(L_DEBUG, "New xdg_shell toplevel title='%s' app_id='%s'",
 		xdg_surface->toplevel->title, xdg_surface->toplevel->app_id);
 	wlr_xdg_surface_ping(xdg_surface);
-	wlr_xdg_toplevel_set_maximized(xdg_surface, true);
+	wlr_xdg_toplevel_set_tiled(xdg_surface, WLR_EDGE_LEFT | WLR_EDGE_RIGHT |
+		WLR_EDGE_TOP | WLR_EDGE_BOTTOM);
```

Comment:
We should probably do both

Reply:
Nah, see the wlroots PR. It falls back to `maximized` if the client doesn't support tiling.

Reply:
I don't understand how that's supposed to work.

Reply:
If the client xdg-shell version is >= 2, then it supports tiling. If the client version is 1, then it doesn't support tiling and we make the window maximized instead. That's the best we can do.

Reply:
I see. I think we should be prepared for clients who set their xdg shell version to the latest but who don't bother implementing tiled to be pretty common.

Reply:
I don't think so. GNOME has tiling too (if you move a window near a border of the screen), so apps that don't implement it will look bad on GNOME too. Also, an upside to tiling is that the app has no request to unset it.

Reply:
Aight.